operator/pkg/securedcluster/scanner/auto_sense.go

```diff
@@ -21,14 +30,39 @@ func AutoSenseLocalScannerSupport(ctx context.Context, client ctrlClient.Client,
 		if err != nil {
 			return false, errors.Wrap(err, "detecting presence of a Central CR in the same namespace")
 		}
-		return !siblingCentralPresent, nil
+		isOpenShift, err := isRunningOnOpenShift(ctx, client, s.GetNamespace())
+		if err != nil {
+			return false, errors.Wrap(err, "cannot fetch OpenShift ClusterVersion resource")
+		}
+		enableScanner := isOpenShift && !siblingCentralPresent
+		return enableScanner, nil
 	case platform.LocalScannerComponentDisabled:
 		return false, nil
 	}
 
 	return false, errors.Errorf("invalid spec.scanner.scannerComponent %q", scannerComponent)
 }
 
+func isRunningOnOpenShift(ctx context.Context, client ctrlClient.Client, namespace string) (bool, error) {
+	log := ctrlLog.FromContext(ctx)
+
+	clusterVersion := &osconfigv1.ClusterVersion{}
+	key := ctrlClient.ObjectKey{Namespace: namespace, Name: ClusterVersionDefaultName}
+	err := client.Get(ctx, key, clusterVersion)
+	if err != nil && errorsK8s.IsNotFound(err) {
+		log.Error(err, "OpenShift ClusterVersion not found")
+		return false, err
+	} else if err != nil && strings.Contains(err.Error(), "no matches for kind") {
```

Comment:
This error happens if `ClusterVersion` kind is not defined. For example, this is the error output for vanilla k8s:
```
error: no matches for kind "ClusterVersion" in version "config.openshift.io/v1"
```